assets/src/web-stories-block/block/block-types/latest-stories/edit.js

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { useDebouncedCallback } from 'use-debounce';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useState } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
+import { __ } from '@wordpress/i18n';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import StoriesInspectorControls from '../../components/storiesInspectorControls';
+import StoriesBlockControls from '../../components/storiesBlockControls';
+import StoriesLoading from '../../components/storiesLoading';
+import { FETCH_STORIES_DEBOUNCE, ORDER_BY_OPTIONS } from '../../constants';
+import StoriesPreview from '../../components/storiesPreview';
+import { useConfig } from '../../../../dashboard/app/config';
+
+/**
+ * Module constants
+ */
+const LATEST_STORIES_QUERY = {
+  per_page: 20,
+  _embed: 'author',
+};
+
+/**
+ * LatestStoriesEdit component
+ *
+ * @param {Object}   root0               Component props.
+ * @param {Object}   root0.attributes    Block attributes.
+ * @param {Function} root0.setAttributes Callable function for saving attribute values.
+ *
+ * @return {*} JSX markup for the editor.
+ */
+const LatestStoriesEdit = ({ attributes, setAttributes }) => {
+  const {
+    blockType,
+    viewType,
+    numOfStories,
+    orderByValue,
+    viewAllLinkLabel,
+    authors,
+  } = attributes;
+
+  const { api } = useConfig();
+
+  const [fetchedStories, setFetchedStories] = useState([]);
+  const [isFetchingStories, setIsFetchingStories] = useState([]);
+
+  /**
+   * Fetch stories based on the query.
+   *
+   * @return {void}
+   */
+  const fetchStories = async () => {
+    try {
+      setIsFetchingStories(true);
+      const stories = await apiFetch({
+        path: addQueryArgs(api.stories, LATEST_STORIES_QUERY),
+      });
+
+      if ('undefined' !== typeof stories && Array.isArray(stories)) {
```

Comment:
When would this be undefined or not an array?

Reply:
it won't be undefined as we are waiting, and the check for the `Array.isArray()` in place to ensure that we work with the array value only.